### ISSUE TITLE
NPE in notifications

### ISSUE THREAD

**Issue by mzorz** [wordpress-mobile/WordPress-Android#6761]
First reported by @daniloercoli in https://github.com/wordpress-mobile/WordPress-Android/pull/6755#issuecomment-338146930
- Just added a new media (the first media of a new post) and got the following crash:
```
10-20 10:40:25.295 29272-29272/org.wordpress.android E/AndroidRuntime: FATAL EXCEPTION: main
  Process: org.wordpress.android, PID: 29272
  org.greenrobot.eventbus.EventBusException: Invoking subscriber failed
      at org.greenrobot.eventbus.EventBus.handleSubscriberException(EventBus.java:505)
      at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:487)
      at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:479)
      at org.greenrobot.eventbus.HandlerPoster.handleMessage(HandlerPoster.java:67)
      at android.os.Handler.dispatchMessage(Handler.java:102)
      at android.os.Looper.loop(Looper.java:135)
      at android.app.ActivityThread.main(ActivityThread.java:5221)
      at java.lang.reflect.Method.invoke(Native Method)
      at java.lang.reflect.Method.invoke(Method.java:372)
      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:898)
      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:693)
   Caused by: java.lang.NullPointerException: Attempt to read from field 'int org.wordpress.android.ui.uploads.PostUploadNotifier$NotificationData.notificationErrorId' on a null object reference
      at org.wordpress.android.ui.uploads.PostUploadNotifier.updateNotificationError(PostUploadNotifier.java:210)
      at org.wordpress.android.ui.uploads.UploadService.cancelPostUploadMatchingMedia(UploadService.java:487)
      at org.wordpress.android.ui.uploads.UploadService.onMediaUploaded(UploadService.java:510)
      at java.lang.reflect.Method.invoke(Native Method)
      at java.lang.reflect.Method.invoke(Method.java:372)
      at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:485)
      at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:479) 
      at org.greenrobot.eventbus.HandlerPoster.handleMessage(HandlerPoster.java:67) 
      at android.os.Handler.dispatchMessage(Handler.java:102) 
      at android.os.Looper.loop(Looper.java:135) 
      at android.app.ActivityThread.main(ActivityThread.java:5221) 
      at java.lang.reflect.Method.invoke(Native Method) 
      at java.lang.reflect.Method.invoke(Method.java:372) 
      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:898) 
      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:693) 
```

**Comment by mzorz** [comment]
Note for later: I'm refactoring the UploadNotifier in #6713 and #6735 and this particular NPE will go away with it

**Comment by mzorz** [comment]
Closing as the new code is merged in `develop` through the merger of #6713
